app/src/test/java/org/simple/clinic/settings/SettingsEffectHandlerTest.kt

```diff
@@ -24,12 +27,14 @@ class SettingsEffectHandlerTest {
   private val userSession = mock<UserSession>()
   private val settingsRepository = mock<SettingsRepository>()
   private val uiActions = mock<UiActions>()
+  private val packageManager = mock<PackageManager>()
```

Comment:
We should not be doing this since we should not be mocking types that we do not own, i.e, `PackageManager` is not owned by us and we should not be mocking it.

Can we put this behind a type, say,

```kotlin
class AppVersionFetcher(packageManager) {
  fun appVersion(applicationId: String): String {
    // Do work here
  }
}
```

and provide this as a dependency to the effect handler instead?

Reply:
Yes, this where I was confused about. Whether or not I can mock system classes in tests.